src/reactapp/src/components/checkoutAgreements/components/CheckoutAgreementModal.jsx

```diff
@@ -2,24 +2,37 @@
 import React from 'react';
 import _get from 'lodash.get';
 import { ClipboardCheckIcon } from '@heroicons/react/outline';
-import { bool, func, oneOfType, shape, string } from 'prop-types';
+
 
 import { __ } from '../../../i18n';
+import { getFormikFieldNameById } from '../utility';
+import { CHECKOUT_AGREEMENTS_FORM } from '../../../config';
 import useAgreementAppContext from '../hooks/useAgreementAppContext';
-
-function CheckoutAgreementModal({ agreementId, actions }) {
+import useAgreementModalContext from '../hooks/useAgreementModalContext';
+import useAgreementFormikContext from '../hooks/useAgreementFormikContext';
+function CheckoutAgreementModal() {
   const { checkoutAgreements } = useAgreementAppContext();
-  const modalAgreement = _get(checkoutAgreements, agreementId);
+  const { activeModalId, setActiveModalId } = useAgreementModalContext();
+  const { setFieldValue } = useAgreementFormikContext();
+  const modalAgreement = _get(checkoutAgreements, activeModalId);
   const agreementTitle = _get(modalAgreement, 'name');
   const isContentHtml = _get(modalAgreement, 'isHtml');
   const agreementContent = _get(modalAgreement, 'content');
 
+  const handleAgreeButtonClick = () => {
+    const fieldName = `${CHECKOUT_AGREEMENTS_FORM}.${getFormikFieldNameById(
+      activeModalId
+    )}`;
+    setFieldValue(fieldName, true);
+    setActiveModalId(false);
+  };
+
   return (
     <div
-      className="fixed inset-0 z-10 overflow-y-auto"
-      aria-labelledby="modal-title"
       role="dialog"
       aria-modal="true"
+      aria-labelledby="modal-title"
+      className="fixed inset-0 z-10 overflow-y-auto"
```

Comment:
Was `z-10` enough to put it on top of the menu?

Reply:
the overlapping issue won't happen as we limited the height of the modal i guess.

Reply:
The overlay will be covering the entire screen (the backdrop), so that should be on top of any other element on the page. I'll have a look after it's merged.